packaging/MacOs/build_macos.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2023 UltiMaker
+# Cura is released under the terms of the LGPLv3 or higher.
+
+
+import os
+import argparse  # Command line arguments parsing and help.
+import subprocess
+from pathlib import Path
+
+ULTIMAKER_CURA_DOMAIN = os.environ.get("ULTIMAKER_CURA_DOMAIN", "nl.ultimaker.cura")
+
+def build_pkg(dist_path: str, app_filename: str, component_filename: str, installer_filename: str) -> None:
+    """ Builds and signs the pkg installer.
+
+    @param dist_path: Path to put output pkg in
+    @param app_filename: name of the .app file to bundle inside the pkg
+    @param component_filename: Name of the pkg component package to bundle the app in
+    @param installer_filename: Name of the installer that contains the component package
+    """
+    pkg_build_executable = os.environ.get("PKG_BUILD_EXECUTABLE", "pkgbuild")
+    product_build_executable = os.environ.get("PRODUCT_BUILD_EXECUTABLE", "productbuild")
+    codesign_identity = os.environ.get("CODESIGN_IDENTITY")
+
+    # This builds the component package that contains UltiMaker-Cura.app. This component package will be bundled in a distribution package.
+    pkg_build_arguments = [
+        pkg_build_executable,
+        "--component",
+        f"{dist_path}/{app_filename}",
+        f"{dist_path}/{component_filename}",
+        "--sign", codesign_identity,
+        "--install-location", "/Applications",
+    ]
+    subprocess.run(pkg_build_arguments)
+
+    # This automatically generates a distribution.xml file that is used to build the installer.
+    # If you want to make any changes to how the installer functions, this file should be changed to do that.
+    # TODO: Use --product {property_list_file} to pull keys out of file for distribution.xml. This can be used to set min requirements
+    distribution_creation_arguments = [
+        product_build_executable,
+        "--synthesize",
+        "--package", f"{dist_path}/{component_filename}",  # Package that will be inside installer
```

Comment:
Use `os.path.join` or `Path()` Good practice is to us paths in a system agnostics way. Maybe we need to cross-compile or somebody else has something funky and weird going on.

I don't think this is actual issue for this line and case, but good practices are there for a reason.